tests/eos-test-utils.c

```diff
@@ -0,0 +1,1960 @@
+	       DownloadOrder order,
+	       const gchar *vendor,
+	       const gchar *product,
+	       const gchar *remote_name,
+	       CmdAsyncStuff *updater_cmd,
+	       GError **error)
+{
+  g_autoptr(GKeyFile) updater_config = NULL;
+  g_autoptr(GKeyFile) hw_config = NULL;
+  g_autoptr(GFile) sysroot = get_sysroot_for_client (client_root);
+  g_autoptr(GFile) repo = get_repo_for_sysroot (sysroot);
+  g_autoptr(GFile) updater_dir = get_updater_dir_for_client (client_root);
+
+  updater_config = get_updater_config (order);
+  hw_config = get_hw_config (vendor, product);
+  if (!prepare_updater_dir (updater_dir,
+                            updater_config,
+                            hw_config,
+                            error))
+    return FALSE;
+  if (!spawn_updater_simple (sysroot,
+                             repo,
+                             updater_dir,
+                             remote_name,
+                             updater_cmd,
+                             error))
+    return FALSE;
+
+  return TRUE;
+}
+
+static gboolean
+ensure_ref_in_subserver (const gchar *ref,
+                         EosTestSubserver *subserver)
+{
+  return g_hash_table_lookup_extended (subserver->ref_to_commit, ref, NULL, NULL);
+}
+
+static gboolean
+ensure_vendor_and_product_in_subserver (const gchar *vendor,
+                                        const gchar *product,
+                                        EosTestSubserver *subserver)
+{
+  guint idx;
+
+  for (idx = 0; idx < subserver->devices->len; ++idx)
+    {
+      EosTestDevice *device = EOS_TEST_DEVICE (g_ptr_array_index (subserver->devices, idx));
+
+      if (g_strcmp0 (vendor, device->vendor) == 0 &&
+          g_strcmp0 (product, device->product) == 0)
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
+EosTestClient *
+eos_test_client_new (GFile *client_root,
+		     const gchar *remote_name,
+		     EosTestSubserver *subserver,
+		     const gchar *ref,
+		     const gchar *vendor,
+		     const gchar *product,
+		     GError **error)
+{
+  g_autoptr(EosTestClient) client = NULL;
+
+  if (!ensure_ref_in_subserver (ref, subserver))
+    return FALSE;
+  if (!ensure_vendor_and_product_in_subserver (vendor, product, subserver))
+    return FALSE;
+
+  if (!prepare_client_sysroot (client_root,
+			       remote_name,
+			       subserver->url,
+			       ref,
+			       subserver->keyid,
+			       error))
+    return FALSE;
+
+  if (!copy_extensions (subserver->repo,
+                        client_root,
+                        ref,
+                        error))
+    return FALSE;
+
+  client = g_object_new (EOS_TEST_TYPE_CLIENT, NULL);
+  client->root = g_object_ref (client_root);
+  client->vendor = g_strdup (vendor);
+  client->product = g_strdup (product);
+  client->remote_name = g_strdup (remote_name);
+  client->ostree_path = g_strdup (subserver->ostree_path);
+  return g_steal_pointer (&client);
+}
+
+gboolean
+eos_test_client_run_updater (EosTestClient *client,
```

Comment:
This seems to have an odd name ... the first impression I got was that this was supposed to run the updater, but it just calls `setup_updater`.

Maybe `eos_setup_updater_for_client`?